matrix_sdk/src/client.rs

```diff
@@ -138,6 +137,8 @@ pub struct Client {
     #[cfg(feature = "encryption")]
     /// Lock making sure we're only doing one key claim request at a time.
     key_claim_lock: Arc<Mutex<()>>,
+    pub(crate) members_request_locks: DashMap<RoomId, Arc<Mutex<()>>>,
+    pub(crate) typing_notice_times: DashMap<RoomId, Instant>,
```

Comment:
These two need to be put into an `Arc`, otherwise we'll be cloning the map and end up with multiple versions of the map, this breaks the de-duplication if you clone the room/client when you send out the requests.

The `group_session_locks` field has the same problem, feel free to add it there as well.

Reply:
oh, i didn't add it since `group_session_locks` doesn't use an `arc`. I will add it.